x/committee/legacy/v0_16/migrate.go

```diff
@@ -15,12 +17,176 @@ import (
 	v038upgrade "github.com/cosmos/cosmos-sdk/x/upgrade/legacy/v038"
 	v040upgrade "github.com/cosmos/cosmos-sdk/x/upgrade/types"
 
+	v016bep3types "github.com/kava-labs/kava/x/bep3/types"
+	v016cdptypes "github.com/kava-labs/kava/x/cdp/types"
 	v015committee "github.com/kava-labs/kava/x/committee/legacy/v0_15"
 	v016committee "github.com/kava-labs/kava/x/committee/types"
+	v016hardtypes "github.com/kava-labs/kava/x/hard/types"
 	v015kavadist "github.com/kava-labs/kava/x/kavadist/legacy/v0_15"
 	v016kavadist "github.com/kava-labs/kava/x/kavadist/types"
+	v016pricefeedtypes "github.com/kava-labs/kava/x/pricefeed/types"
 )
 
+// migrateWhitelist returns an string slice of json keys that should be whitelisted on the whitelist interface
+func migrateWhitelist(whitelist interface{}, ignoredTag string) []string {
+	allowed := []string{}
+	v := reflect.ValueOf(whitelist)
+	typeOfS := v.Type()
+	for i := 0; i < v.NumField(); i++ {
+		tag := typeOfS.Field(i).Tag.Get("json")
+		if tag != ignoredTag && tag != "" {
+			val, ok := v.Field(i).Interface().(bool)
+			if ok && val {
+				allowed = append(allowed, tag)
+			}
+		}
+	}
+	sort.Strings(allowed)
+	return allowed
+}
+
+// isSubparamAllowed returns true if the subspace and key is allowed in the v15 permissions
+func isSubparamAllowed(permission v015committee.SubParamChangePermission, subspace string, key string) bool {
+	for _, allowed := range permission.AllowedParams {
+		if allowed.Key == key && allowed.Subspace == subspace {
+			return true
+		}
+	}
+	return false
+}
+
+type subspaceKeyPair struct {
+	key      []byte
+	subspace string
+}
+
+// migrateSubParamPermissions converts v15 SubParamChangePermissions to v16 ParamsChangePermission
+func migrateSubParamPermissions(permission v015committee.SubParamChangePermission) *v016committee.ParamsChangePermission {
+	changes := v016committee.AllowedParamsChanges{}
+
+	// migrate allowed params
+	pairsToAvoid := []subspaceKeyPair{
```

Comment:
avoiding a couple params here since we are handling them later and the new subparam permission does not need them to be recorded twice.